### PR TITLE
added libguarded

### PR DESCRIPTION
**Pull request by qchateau**
Specify library name and version:  **libguarded/1.1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libguarded/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            self.run(os.path.join("bin", "main"))
```
Comment (madebr) [inline recipes/libguarded/all/test_package/conanfile.py]:
```suggestion
            self.run(os.path.join("bin", "main"), run_environment=True)
```

File: recipes/libguarded/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
+
+class LibguardedConan(ConanFile):
+    name = "libguarded"
```
Comment (uilianries) [inline recipes/libguarded/all/conanfile.py]:
```suggestion
    name = "cs_libguarded"
```
I would prefer `cs_libguarded`. Both Cmake project name and github project are the same.

File: recipes/libguarded/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/copperspice/libguarded"
+    description = "The libGuarded library is a standalone header-only library for multithreaded programming."
+    topics = ("multithreading",)
```
Comment (uilianries) [inline recipes/libguarded/all/conanfile.py]:
```suggestion
    topics = ("multithreading", "templates", "cpp14", "mutexes")
```

File: recipes/libguarded/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "cs_libguarded-libguarded-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        self.copy("*.hpp", src=os.path.join(self._source_subfolder, "src"), dst='include')
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_id(self):
```
Comment (uilianries) [inline recipes/libguarded/all/conanfile.py]: `self.info.header_only()` is not required if you don't have settings in your recipe

File: recipes/libguarded/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(main main.cpp)
+target_link_libraries(main CONAN_PKG::libguarded)
+set_property(TARGET main PROPERTY CXX_STANDARD 11)
```
Comment (uilianries) [inline recipes/libguarded/all/test_package/CMakeLists.txt]:
```suggestion
set_property(TARGET main PROPERTY CXX_STANDARD 14)
```

It's c++14 according the project page

Comment (qchateau) [reply recipes/libguarded/all/test_package/CMakeLists.txt]: According to the [documentation](https://www.copperspice.com/docs/cs_libguarded/lib-api.html), part of the library is C++11, but you need C++14 to be able to use all of it. As the test_package shows, it is perfectly fine to use the core feature of the library with C++11.

Not sure how to reflect that on the project tags

Comment (uilianries) [reply recipes/libguarded/all/test_package/CMakeLists.txt]: Agreed!

File: recipes/cs_libguarded/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
```
Comment (uilianries) [inline recipes/cs_libguarded/all/test_package/CMakeLists.txt]: cmake targets requires CMake 3+. I would prefer keeping cmake 2.8 and using ${CONAN_LIBS} instead.

**Pull request merged**